api/lib/domain/services/scoring/scoring-service.js

```diff
@@ -0,0 +1,73 @@
+const {
+  PIX_COUNT_BY_LEVEL,
+  MAX_REACHABLE_LEVEL,
+  MAX_REACHABLE_PIX_BY_COMPETENCE,
+} = require('../../constants');
+
+const _ = require('lodash');
+
+function calculateScoringInformationForCompetence(knowledgeElements, blockReachablePixAndLevel = true) {
+
+  const realTotalPixScoreForCompetence = _(knowledgeElements).sumBy('earnedPix');
+  const pixScoreForCompetence = _getPixScoreForOneCompetence(realTotalPixScoreForCompetence, blockReachablePixAndLevel);
+  const currentLevel = _getCompetenceLevel(pixScoreForCompetence, blockReachablePixAndLevel);
+  const pixAheadForNextLevel = _getPixScoreAheadOfNextLevel(pixScoreForCompetence);
+  return {
+    realTotalPixScoreForCompetence,
+    pixScoreForCompetence,
+    currentLevel,
+    pixAheadForNextLevel
+  };
+}
+
+function getBlockedLevel(level) {
+  return _getBlockedLevel(level);
+}
+function getCompetencePixScore(pixScore) {
+  return _getBlockedPixScoreByCompetence(pixScore);
+}
+
+function getCompetenceLevelByPix(pix) {
+  return _getCompetenceLevel(pix, true);
+}
+
+function totalUserPixScore(pixEarnedByCompetence) {
+  const pixByCompetenceLimited = _.map(pixEarnedByCompetence, (pixEarnedForOneCompetence) => _getBlockedPixScoreByCompetence(pixEarnedForOneCompetence));
+  return _.sum(pixByCompetenceLimited);
+}
+
+module.exports = {
+  calculateScoringInformationForCompetence,
+  getBlockedLevel,
+  getCompetenceLevelByPix,
+  totalUserPixScore,
+  getCompetencePixScore
+};
+
+function _getPixScoreForOneCompetence(exactlyEarnedPix, blockReachablePixAndLevel = true) {
+  const userEarnedPix = _.floor(exactlyEarnedPix);
+  if (blockReachablePixAndLevel) {
+    return _getBlockedPixScoreByCompetence(userEarnedPix);
+  }
+  return userEarnedPix;
+}
+
+function _getCompetenceLevel(pixScoreForCompetence, blockReachablePixAndLevel = true) {
```

Comment:
je suis pas fan des paramètres booléens
a minima, je pense qu'il faudrait utiliser un objet
```suggestion
function _getCompetenceLevel({ pixScoreForCompetence, blockReachablePixAndLevel = true }) {
```